server/channels/store/sqlstore/sqlx_wrapper.go

```diff
@@ -249,12 +249,19 @@ func (w *sqlxDBWrapper) SelectCtx(ctx context.Context, dest any, query string, a
 }
 
 func (w *sqlxDBWrapper) SelectBuilder(dest any, builder Builder) error {
+	return w.SelectBuilderCtx(context.Background(), dest, builder)
+}
+
+func (w *sqlxDBWrapper) SelectBuilderCtx(ctx context.Context, dest any, builder Builder) error {
 	query, args, err := builder.ToSql()
 	if err != nil {
 		return err
 	}
 
-	return w.Select(dest, query, args...)
+	ctx, cancel := context.WithTimeout(ctx, w.queryTimeout)
+	defer cancel()
+
+	return w.SelectCtx(ctx, dest, query, args...)
```

Comment:
Adding a timeout as a side effect is unexpected. I would have expected `*Ctx` variants to just pass through a context that has already been configured.